### ISSUE TITLE
discord down again 

### ISSUE THREAD

**Issue by ProgrammerIn-wonderland** [immortal-miku/talk-with-me#5]

**Comment by ProgrammerIn-wonderland** [comment]
@nylecohen

**Comment by NyleCohen** [comment]
this is discord but built different

**Comment by NyleCohen** [comment]
![image](https://user-images.githubusercontent.com/28657776/138611436-a16fb519-c1d9-4e56-aee7-dbe2f9a2b576.png)

**Comment by ProgrammerIn-wonderland** [comment]
ikr

**Comment by ProgrammerIn-wonderland** [comment]
I cant believe they made github issues real time

**Comment by NyleCohen** [comment]
its incredible

**Comment by ProgrammerIn-wonderland** [comment]
yeah

**Comment by ProgrammerIn-wonderland** [comment]
discord is having a stroke lmfao

**Comment by ProgrammerIn-wonderland** [comment]
as I was saying the fold3 loses cameras when you unlock bootloader

**Comment by ProgrammerIn-wonderland** [comment]
I think thats just bull shit

**Comment by NyleCohen** [comment]
should of stuck with go discord devs

**Comment by NyleCohen** [comment]
i cant imagine it looses camera

**Comment by NyleCohen** [comment]
does samsung hate us that much to lock down the fucking camera?

**Comment by ProgrammerIn-wonderland** [comment]
you dont have to imagine because https://www.xda-developers.com/samsung-galaxy-z-fold-3-unlock-bootloader-broken-camera/

**Comment by NyleCohen** [comment]
thats fucking dumb
but i probably wouldnt unlock the bootloader on a device that expensive

**Comment by ProgrammerIn-wonderland** [comment]
then how do you root?

**Comment by NyleCohen** [comment]
i dont

**Comment by ProgrammerIn-wonderland** [comment]
so you want a device you dont fully own?

**Comment by NyleCohen** [comment]
no, but thats the world we live in right now

**Comment by ProgrammerIn-wonderland** [comment]
I wish one day linux phoness aren't  shit

**Comment by NyleCohen** [comment]
i hope thats soon i would buy one so fast
we need an actual good mobile linux os that is backed by a good corporation, ubports just isnt enough for ut rn considering its still on 16.04

**Comment by ProgrammerIn-wonderland** [comment]
SIXTEEN  DOT FUCKING FOUR?  THATS 2016 APRIL HOLY  FUCK THATS  OLD

**Comment by NyleCohen** [comment]
exactly, one of the things holding it back, they just dont have nearly enough developers or support to update it
im pretty sure it hasnt moved from 16.04 since canonical abandoned it

**Comment by NyleCohen** [comment]
im getting so many dm's on discord and signal asking me "why is discord down" I DONT FUCKING KNOW ASK JASON CITRON OR WHATEVER

**Comment by ProgrammerIn-wonderland** [comment]
ok so I  found something really cool  in kde  @NyleCohen

**Comment by NyleCohen** [comment]
tell me im gonna grab a drink real quick

**Comment by ProgrammerIn-wonderland** [comment]
![Screenshot_20211024_163510](https://user-images.githubusercontent.com/30693865/138611906-c92fd7da-f83b-495d-8ab9-81226d6e845c.png)
the usual app switcher is this

**Comment by ProgrammerIn-wonderland** [comment]
you get it by doing alt+tab

**Comment by ProgrammerIn-wonderland** [comment]
turns out if you do cntrl tab

**Comment by ProgrammerIn-wonderland** [comment]
nvm  it was a  firefox thing

**Comment by ProgrammerIn-wonderland** [comment]
![image](https://user-images.githubusercontent.com/30693865/138611988-a10fe84b-940f-4d41-9793-6f431e6ee8c7.png)
still cool tho

**Comment by NyleCohen** [comment]
firefox changes your control tab?
why

**Comment by NyleCohen** [comment]
hey you cant do this in discord

# WHY

**Comment by ProgrammerIn-wonderland** [comment]
firefox makes cntrl tab like alt tab but for browser

**Comment by ProgrammerIn-wonderland** [comment]
> hey you cant do this in discord
> # WHY

markdown is cool

**Comment by NyleCohen** [comment]
it kinda works in vivaldi but with no tab ui so its a bit shit
just use ctrl+<tab number>

**Comment by ProgrammerIn-wonderland** [comment]
you also cant do this [https://github.com/NyleCohen](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

**Comment by NyleCohen** [comment]
# FUCK YOU

**Comment by ProgrammerIn-wonderland** [comment]
actually,  you can

**Comment by ProgrammerIn-wonderland** [comment]
but only bots can do it

**Comment by NyleCohen** [comment]
wait really?
thats kinda sick

**Comment by ProgrammerIn-wonderland** [comment]
yeah, human users cant do markdown links

**Comment by ProgrammerIn-wonderland** [comment]
I've rick rolled a few people using a roleplay bot

**Comment by NyleCohen** [comment]
thats actually pretty cool

![image](https://user-images.githubusercontent.com/28657776/138612195-e043f6a3-a249-4aee-95b4-773f5ffccb8f.png)
RIP WSL

**Comment by ProgrammerIn-wonderland** [comment]
hyperV just fuckin died lmfaoooo

**Comment by NyleCohen** [comment]
shoulda used qemu

**Comment by ProgrammerIn-wonderland** [comment]
![Screenshot_20211024_164352](https://user-images.githubusercontent.com/30693865/138612233-1b29cef8-3873-44a6-ab12-feb1cef23b22.png)
github rich presence when?

**Comment by NyleCohen** [comment]
also something i found pretty annoying while using my new mac, couldnt get virt-manager working for a debian vm

**Comment by ProgrammerIn-wonderland** [comment]
imagine someone rejecting your PR because of your music taste

**Comment by NyleCohen** [comment]
THEY GOTTA ADD RPC TO GITHUB DUDE
wait yknow the speech bubble thing you can put on your profile, you could automate it for spotify?????

**Comment by ProgrammerIn-wonderland** [comment]
you probably could

**Comment by NyleCohen** [comment]
i might write that now

**Comment by ProgrammerIn-wonderland** [comment]
wait let me look at github  api endpoints

**Comment by ProgrammerIn-wonderland** [comment]
![image](https://user-images.githubusercontent.com/30693865/138612323-dba98f04-c0c8-4ed3-800a-37bafb36d42a.png)
omg

**Comment by NyleCohen** [comment]
no way
what if you could pull discord statuses and update it to your github profile

**Comment by ProgrammerIn-wonderland** [comment]
why is this unironically better than discord api
https://docs.github.com/en/rest

**Comment by NyleCohen** [comment]
ive never used gh api before tbf
let me see if theres any libs with the user data endpoint implemented

**Comment by ProgrammerIn-wonderland** [comment]
it looks simple as hell

**Comment by NyleCohen** [comment]
fuck this
`npm init`

**Comment by ProgrammerIn-wonderland** [comment]
using this lib (official) [Octokit](https://github.com/octokit/core.js#readme)
the code is
```js
await octokit.request('PATCH /user', {
  bio: 'new status'
})
```

**Comment by NyleCohen** [comment]
they call it octokit thats so sick

**Comment by ProgrammerIn-wonderland** [comment]
![image](https://user-images.githubusercontent.com/30693865/138612487-d3b12cf4-be0a-4164-9149-d1201e095d78.png)

**Comment by NyleCohen** [comment]
> using this lib (official) [Octokit](https://github.com/octokit/core.js#readme) the code is
> 
> ```js
> await octokit.request('PATCH /user', {
>   bio: 'new status'
> })
> ```

thats dead fucking simple

**Comment by ProgrammerIn-wonderland** [comment]
why aren't all libs like this

**Comment by ProgrammerIn-wonderland** [comment]
(and APIs)

**Comment by NyleCohen** [comment]
because developers are dumb
bear with me if i take a while to reply im multitasking with like 4 other things

**Comment by ProgrammerIn-wonderland** [comment]
I understand lol

**Comment by ProgrammerIn-wonderland** [comment]
> no way what if you could pull discord statuses and update it to your github profile

either requires self bot or for you to be in a discord server with a bot checking for status updates (priviledged intent)

**Comment by NyleCohen** [comment]
> > no way what if you could pull discord statuses and update it to your github profile
> 
> either requires self bot or for you to be in a discord server with a bot checking for status updates (priviledged intent)

yeah i feel like thats probably overkill, spotify integration could be cool tho, i would totally just leave it in a container

**Comment by ProgrammerIn-wonderland** [comment]
yeah it would be

**Comment by NyleCohen** [comment]
although last time i used spotifyapi i wanted to throw my keyboard against a wall, hopefully its a little easier this time

**Comment by ProgrammerIn-wonderland** [comment]
I'm kinda surprised how good the github api is tho

**Comment by ProgrammerIn-wonderland** [comment]
everything is exposed

**Comment by NyleCohen** [comment]
i kinda want to fuck with it now, ive never used it before

**Comment by ProgrammerIn-wonderland** [comment]
same

**Comment by NyleCohen** [comment]
![image](https://user-images.githubusercontent.com/28657776/138612726-2e6abc20-0283-42f9-9ae0-7c40a093a1de.png)
is it me or is this new

**Comment by ProgrammerIn-wonderland** [comment]
I think thats new

**Comment by ProgrammerIn-wonderland** [comment]
![image](https://user-images.githubusercontent.com/30693865/138612810-4e2b63bb-419b-46e9-a440-0783ed729f31.png)

**Comment by ProgrammerIn-wonderland** [comment]
![image](https://user-images.githubusercontent.com/30693865/138612816-a2488243-505b-4cbd-8713-f6ca46a21387.png)

**Comment by ProgrammerIn-wonderland** [comment]
seems like bio and whatever the fuck status is, are different

**Comment by NyleCohen** [comment]
nice, gonna try it once i get this done
can i see your status?

**Comment by NyleCohen** [comment]
![image](https://user-images.githubusercontent.com/28657776/138612875-e6308fed-1ea5-45e3-af03-c0561878c798.png)
i can

**Comment by ProgrammerIn-wonderland** [comment]
> nice, gonna try it once i get this done can i see your status?

here is shell code (curl)
```bash
curl \
  -X PATCH \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/user \
  -d '{"bio":"test"}' \
 -u "NyleCohen"
```
when it asks for password dont put  your github password
go here https://github.com/settings/tokens/new 
select this
![image](https://user-images.githubusercontent.com/30693865/138612949-61d77a6e-90a5-4ce1-82ee-ad0a723f7653.png)
and use the token

**Comment by NyleCohen** [comment]
agh ik personal access tokens are a 1000x more secure but i just cant use them well

**Comment by ProgrammerIn-wonderland** [comment]
same

**Comment by ProgrammerIn-wonderland** [comment]
kinda sad about the password deprecation, so what if I hate security?

**Comment by ProgrammerIn-wonderland** [comment]
if I'm insecure in real life let me be insecure online too god damn it

**Comment by NyleCohen** [comment]
hahaha true
to be fair im probably the most insecure person online which is super ironic considering i preach internet security all the time
literally 2 days ago my vrchat account got a login from the united states, i had to change my password

**Comment by ProgrammerIn-wonderland** [comment]
aww its not real time on phone

**Comment by NyleCohen** [comment]
fuck
come on github you were on to something here

**Comment by ProgrammerIn-wonderland** [comment]
test

**Comment by NyleCohen** [comment]
test passed

**Comment by ProgrammerIn-wonderland** [comment]
Ok the mobile website is real time

**Comment by ProgrammerIn-wonderland** [comment]
The mobile app isnt

**Comment by NyleCohen** [comment]
ok cool
wait i wonder if fasthub is realtime let me check

**Comment by ProgrammerIn-wonderland** [comment]
Hey at least they have a mobile website unlike discord

**Comment by NyleCohen** [comment]
lmaooo true
then again most of the user base doesnt use web discord so ig there isnt a reason

**Comment by NyleCohen** [comment]
Test

**Comment by ProgrammerIn-wonderland** [comment]
Here is the discord "mobile website"
![Screenshot_20211024-171404_Kiwi_Browser](https://user-images.githubusercontent.com/30693865/138613191-3564c619-639c-44eb-9a1e-c8c67a2c9fef.png)

**Comment by ProgrammerIn-wonderland** [comment]
I think the reason the mobile app isnt real time is because the github app uses the actual github api (thats also probably why the api is so good) and the api doesn't have realtime issues

**Comment by NyleCohen** [comment]
fasthub isnt real time 😢 
yeah that would make sense actually, same with fasthub
GITHUB REAL TIME ISSUES API WHEN

**Comment by ProgrammerIn-wonderland** [comment]
I'm dead wrong github realtime issues have an api endpoint https://docs.github.com/en/developers/webhooks-and-events/webhooks/about-webhooks#events

**Comment by NyleCohen** [comment]
no way
it has a webhook
im gonna fork fasthub and implement this

**Comment by ProgrammerIn-wonderland** [comment]
Here is the actual doc for it sir

https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#issues

Also what's fasthub

**Comment by NyleCohen** [comment]
> Here is the actual doc for it sir

thank you
> Also what's fasthub

open source github android client ive been using for years before the github app was made

**Comment by ProgrammerIn-wonderland** [comment]
Ok so while it is real time it requires you to have a real server to like... Yknow, receive the webhook

**Comment by ProgrammerIn-wonderland** [comment]
Not as good as the online version but its something

**Comment by NyleCohen** [comment]
true true
you could make a chat app that just uses gh issues as a backend
that could be really funny

**Comment by ProgrammerIn-wonderland** [comment]
Very funny indeed

**Comment by ProgrammerIn-wonderland** [comment]
But a better bet is github discussions ;)

**Comment by ProgrammerIn-wonderland** [comment]
We should implement an unofficial api for the websocket realtime issues

**Comment by NyleCohen** [comment]
that could be kinda cool

**Comment by ProgrammerIn-wonderland** [comment]
I'm.gonna go now I needa do homework, cya later!

**Comment by NyleCohen** [comment]
cya
this was a fucking experience lmao

**Comment by ProgrammerIn-wonderland** [comment]
Yesss
